source/common/stream_info/filter_state_impl.cc

```diff
@@ -79,6 +79,26 @@ FilterState::Object* FilterStateImpl::getDataMutableGeneric(absl::string_view da
   return current->data_.get();
 }
 
+std::shared_ptr<FilterState::Object>
+FilterStateImpl::getDataSharedMutableGeneric(absl::string_view data_name) {
```

Comment:
This is the same function as above, right? Can't you just implement the one above in terms of this one?

Reply:
No, it has a different recursive call. Not sure how to templatize the recursion so copying seems simpler.

Reply:
I'm probably missing something, but can't you just call the shared_ptr version and then just convert it to raw pointer after the recursion is complete?

Reply:
Done.